libs/pxt-common/pxt-helpers.ts

```diff
@@ -201,6 +201,63 @@ namespace helpers {
 
         return s.substr(start, end - start);
     }
+
+    export function stringSplit(S: string, separator?: string, limit?: number): string[] {
+        // https://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.split
+        const A: string[] = [];
+        let lim = 0;
+        if (limit === undefined)
+            lim = (1 << 30) - 1; // spec says 1 << 53
+        else if (limit < 0)
+            lim = 0;
+        else
+            lim = limit | 0;
+        const s = S.length;
+        let p = 0;
+        const R = separator;
+        if (lim == 0)
+            return A;
+        if (separator === undefined) {
+            A[0] = S;
+            return A;
+        }
+        if (s == 0) {
+            let z = splitMatch(S, 0, R);
+            if (z > -1) return A;
+            A[0] = S;
+            return A;
+        }
+        let T: string;
+        let q = p;
+        while (q != s) {
+            let e = splitMatch(S, q, R);
```

Comment:
using `indexOf` would be faster, though maybe a bit more complicated

Reply:
I was really just following the spec. We should do a deeper pass on JS support for string/arrays.